src/lava/lib/optimization/solvers/generic/scif/process.py

```diff
@@ -21,20 +21,38 @@ def __init__(
             self,
             *,
             shape: ty.Tuple[int, ...],
-            step_size: ty.Optional[int] = 1,
+            step_size: ty.Optional[ty.Union[int, npty.NDArray]] = 1,
             theta: ty.Optional[int] = 4,
-            noise_amplitude: ty.Optional[int] = 0) -> None:
+            sustained_on_tau: ty.Optional[int] = 0,
+            noise_amplitude: ty.Optional[int] = 0,
+            noise_precision: ty.Optional[int] = 0) -> None:
         """
         Stochastic Constraint Integrate and Fire neuron Process.
+
         Parameters
         ----------
         shape: Tuple
             Number of neurons. Default is (1,).
         step_size: int
-            bias current driving the SCIF neuron. Default is 1 (arbitrary).
+            The bias driving a SCIF neuron. Default is arbitrarily chosen to
+            be 1. In the case of quadratic unconstrained binary optimization
+            (QUBO) problems, `step_size` is the vector formed by the diagonal
+            elements of the cost matrix.
```

Comment:
We should actually send this step size as a multiplicative factor for the gradient descent. In QUBO the matrix diagonal has to match the input problem so we cannot allow its deformation. IF that's delayed into a follow up PR, then we should add a NotImplemented error if the user sets it so that the user get's notified step_size will not have any effect for now.